docs/NooBaaNonContainerized/Configuration.md

```diff
@@ -0,0 +1,321 @@
+
+`system.json` - 
+* <u>Type</u>: File.
+* <u>Required</u>: True.
+* <u>Description</u>: A JSON file that contains information about the system deployed on the machine. 
+* <u>Example</u>:
+    ```JSON
+    { 
+        "hostname1": { 
+            "current_version":"5.17.0",
+            "upgrade_history":{
+                "successful_upgrades":[]
+            }
+        },
+        "hostname2": { 
+            "current_version":"5.17.0",
+            "upgrade_history":{
+                "successful_upgrades":[]
+            }
+        }
+    }
+    ```
+`accounts/` - 
+* <u>Type</u>: Directory.
+* <u>Required</u>: True.
+* <u>Description</u>: A directory that contains configuration files for individual accounts, each account configuration file is named {account_name}.json and adheres to the [account schema](../../src/server/system_services/schemas/nsfs_account_schema.js).
+* <u>Example</u>:
+    ```sh
+    > ls /etc/noobaa.conf.d/accounts/
+    alice.json
+    bob.json
+    charlie.json
+    ```
+
+`access_keys/` 
+* <u>Type</u>: Directory.
+* <u>Required</u>: True.
+* <u>Description</u>: A directory that contains symlinks to accounts configurations, each symlink named {access_key}.symlink, linking to an account within `accounts/` directory. The access key symlink points to a relative path of the account rather than an absolute path, for example: `../accounts/alice.json`.
+* <u>Example</u>:
+    ```sh
+    > ls -la /etc/noobaa.conf.d/access_keys/
+    ABCABCABCABCABCABCAB.symlink -> ../accounts/alice.json
+    DEFDEFDEFDEFDEFDEFDE.symlink -> ../accounts/bob.json
+    GHIGHIGHIGHIGHIGHIGH.symlink -> ../accounts/charlie.json
+    ```
+`buckets/`
+* <u>Type</u>: Directory.
+* <u>Required</u>: True.
+* <u>Description</u>: A directory that contains configuration files for individual buckets, each bucket configuration file is named {bucket_name}.json and adheres to the [bucket schema](../../src/server/system_services/schemas/nsfs_bucket_schema.js).
+* <u>Example</u>:
+    ```sh
+    > ls /etc/noobaa.conf.d/buckets/
+    bucket1.json
+    bucket2.json
+    bucket3.json
+    ```
+`config.json`
+* <u>Type</u>: File.
+* <u>Required</u>: False.
+* <u>Description</u>: A JSON file that contains shared configurations of all nodes on the cluster and machine specific configurations, for more details check - [Config File Customizations](#config-file-customizations)
+* <u>Example</u>: 
+    ```JSON
+    {
+        "ENDPOINT_FORKS": 2,
+        "host_customization": {
+            "{node_name}" : {
+                "ENDPOINT_FORKS": 3, 
+            }
+        }
+    }
+    ```
+
+`master_keys.json`
+* <u>Type</u>: File.
+* <u>Required</u>: False.
+* <u>Description</u>: A JSON file containing master keys information. The master keys specified in this file used for encryption of sensitive data, such as account's secret key. When using 'executable' master keys store type, the master_keys.json is stored on a different location, therefore this file is optional.
```

Comment:
Don't we have more details about it (and add a link?)
Some questions are raised here - why will I use 'executable'? What are the existing options? (file...).